### PR TITLE
Dependency updates.

### PR DESCRIPTION
**Pull request by grundleborg**
#### Summary
Dependency Updates

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-21581

### CODE REVIEW

**Comment by grundleborg** [comment]
@agnivade @mgdelacroix requesting a quick re-review as I updated this PR since it's been idle for ages. This time I'm going to request QA to look at it urgently so we can get it merged before it goes stale.

File: config/database_test.go [hunk 1]
```diff
@@ -701,7 +701,7 @@ func TestDatabaseStoreLoad(t *testing.T) {
 		assert.Equal(t, map[string]interface{}{"SqlSettings": map[string]interface{}{"DataSourceReplicas": true}}, ds.GetEnvironmentOverrides())
 		// check that in DB config does not include overwritten variable
 		_, actualConfig := getActualDatabaseConfig(t)
-		assert.Equal(t, []string(nil), actualConfig.SqlSettings.DataSourceReplicas)
+		assert.Equal(t, []string{}, actualConfig.SqlSettings.DataSourceReplicas)
```
Comment (grundleborg) [inline config/database_test.go:704]: @cpoile is this change correct? I needed it to get the tests passing, but honestly not sure if the change here means the test should actually be failing?

File: config/file_test.go [hunk 2]
```diff
@@ -660,7 +660,7 @@ func TestFileStoreLoad(t *testing.T) {
 		assert.Equal(t, map[string]interface{}{"SqlSettings": map[string]interface{}{"DataSourceReplicas": true}}, fs.GetEnvironmentOverrides())
 		// check that on disk config does not include overwritten variable
 		actualConfig := getActualFileConfig(t, path)
-		assert.Equal(t, []string(nil), actualConfig.SqlSettings.DataSourceReplicas)
+		assert.Equal(t, []string{}, actualConfig.SqlSettings.DataSourceReplicas)
```
Comment (grundleborg) [inline config/file_test.go:663]: @cpoile same question here?

**Comment by grundleborg** [comment]
@cpoile adding a request for your review for two changes to tests specifically that it looks like you wrote.

File: config/database_test.go (hunk 1 above)
Comment (cpoile) [reply config/database_test.go:704]: Yep, that's good -- the test should pass (the original DB config should have an empty slice in `DataSourceReplicas`).

File: config/file_test.go (hunk 2 above)
Comment (cpoile) [reply config/file_test.go:663]: Same as above.

**Comment by agnivade** [comment]
Sorry @grundleborg, I merged my auto-vendoring changes which has created conflicts now. To fix it, you can just overwrite my changes on top of your vendor/modules.txt, and then re-create the file again by doing "go mod vendor". Everything should be same, except some more "#explicit" directives should get added. Sorry for the troubles :face_with_head_bandage: :weary:

**Comment by grundleborg** [comment]
:imp:

**Comment by grundleborg** [comment]
@agnivade does this look OK now?

**Comment by agnivade** [comment]
Looks great, thanks !

**Pull request merged**